src/components/TypeWriter.jsx

```diff
@@ -44,8 +44,10 @@ class TypeWriter extends React.Component {
   }
 
   shouldComponentUpdate(nextProps, nextState) {
+    const visibleCharsChanged = this.state.visibleChars !== nextState.visibleChars;
+    const childrenChanged = this.props.children !== nextProps.children;
```

Comment:
I don't think `childrenChanged` will ever be `false` since we are comparing two non-primitives here. This change will just force the component to be updated all the time.

You probably need React's children utilities (https://facebook.github.io/react/docs/top-level-api.html#react.children) to make assumptions about the children.